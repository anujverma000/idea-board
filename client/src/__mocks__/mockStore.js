import configureStore from 'redux-mock-store'
import thunk from "redux-thunk";
import rootReducer from "../reducers";

const initialState= {
    idea: {
        list: []
    }
}
const mockStore = configureStore([thunk]);
const store = mockStore(initialState);
store.replaceReducer(rootReducer);

export default store;