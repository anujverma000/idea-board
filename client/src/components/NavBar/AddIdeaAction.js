import React from "react";
import { connect } from 'react-redux';
import styled from "styled-components";
import { createIdea } from '../../actions/ideaActions';

const AddIdeaAction = ({ createNewIdea }) => {
    return(
        <Wrapper onClick={createNewIdea} >Add new idea</Wrapper>
    )
}


/**
 *  STYLES
 */
const Wrapper = styled.button`
    color: ${props => props.theme.textColorPrimary};
    padding: 16px 24px;
    font-size: 16px;
    border-radius: 4px;
    cursor: pointer;
    box-sizing: border-box;
    min-width: 150px;
    text-align: center;
    border: 1px solid ${props => props.theme.textColorPrimary};
    background-color: ${props => props.theme.backgroundColor};
    &:hover{
        transition: .3s;
        color: ${props => props.theme.textColorInvert};
        background-color: ${props => props.theme.primaryColorLight};
    }
    &:active{
        color: ${props => props.theme.primaryColor};
    }
`;

const mapDispatchToProps = dispatch => ({
    createNewIdea: () =>  dispatch(createIdea())
});

export default connect(null ,mapDispatchToProps)(AddIdeaAction)