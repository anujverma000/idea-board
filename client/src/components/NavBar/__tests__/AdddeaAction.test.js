import React from 'react';
import renderer from 'react-test-renderer';
import AddIdeaAction from '../AddIdeaAction'
import mockStore from "../../../__mocks__/mockStore";
import { Provider } from "react-redux";

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={mockStore}><AddIdeaAction /></Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
