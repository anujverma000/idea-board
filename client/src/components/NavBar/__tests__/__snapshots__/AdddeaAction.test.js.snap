// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="sc-bdVaJa clmorv"
  onClick={[Function]}
>
  Add new idea
</button>
`;
